src/display/canvas.js

```diff
@@ -1004,6 +1004,10 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
       composeSMask(this.ctx, this.current.activeSMask, groupCtx);
       this.ctx.restore();
       copyCtxState(groupCtx, this.ctx);
+
+      if (this.baseTransform) {
+        this.ctx.setTransform.apply(this.ctx, this.baseTransform);
```

Comment:
this.ctx.restore(); shall do that beginGroup and endGroup shall restore mozCurrentTransform. All is happening now that we set transform to (1, 0, 0, 1, 0, 0) state in most of the cases, and this confuses me. SMask is applied by setGState command in-between save/restore so that shall not trigger any transforms. I wonder if this.baseTransform is equal to offset mozCurrentTransform of the groupCtx.